javalib/src/main/scala/java/time/YearMonth.scala

```diff
@@ -0,0 +1,246 @@
+      else ValueRange.of(1, Year.MAX_VALUE)
+
+    case _ => super.range(field)
+  }
+
+  def getLong(field: TemporalField): Long = field match {
+    case YEAR            => year
+    case YEAR_OF_ERA     => if (year < 1) 1 - year else year
+    case MONTH_OF_YEAR   => month
+    case PROLEPTIC_MONTH => prolepticMonth
+    case ERA             => if (year < 1) 0 else 1
+
+    case _: ChronoField  =>
+      throw new UnsupportedTemporalTypeException("Unsupported field: " + field)
+
+    case _ => field.getFrom(this)
+  }
+
+  private def prolepticMonth: Long = year * 12L + month - 1
+
+  def getYear(): Int = year
+
+  def getMonthValue(): Int = month
+
+  def getMonth(): Month = Month.of(month)
+
+  def isLeapYear(): Boolean = Year.isLeap(year)
+
+  def isValidDay(dayOfMonth: Int): Boolean =
+    dayOfMonth >= 1 && dayOfMonth <= lengthOfMonth()
+
+  def lengthOfMonth(): Int = getMonth().length(isLeapYear())
+
+  def lengthOfYear(): Int = Year.of(year).length()
+
+  def `with`(field: TemporalField, value: Long): YearMonth = {
+    def withYearMonth(y: Int, m: Int): YearMonth = {
+      if (y == year && m == month) this
+      else YearMonth.of(y, m)
+    }
+
+    field match {
+      case YEAR =>
+        withYearMonth(YEAR.checkValidIntValue(value), month)
+
+      case YEAR_OF_ERA =>
+        val isoYear = YEAR.checkValidIntValue(
+          if (year < 1) 1 - value
+          else value
+        )
+        withYearMonth(isoYear, month)
+
+      case MONTH_OF_YEAR =>
+        withYearMonth(year, MONTH_OF_YEAR.checkValidIntValue(value))
+
+      case PROLEPTIC_MONTH =>
+        plusMonths(value - getLong(PROLEPTIC_MONTH))
+
+      case ERA =>
+        requireDateTime(value >= 0 && value <= 1, s"Invalid value for ERA: $value")
+        val era = getLong(ERA)
+        if (value == era) this
+        else withYearMonth(YEAR.checkValidIntValue(1 - year), month)
+
+      case _: ChronoField =>
+        throw new UnsupportedTemporalTypeException("Unsupported field: " + field)
+
+      case _ => field.adjustInto(this, value)
+    }
+  }
+
+  def withYear(year: Int): YearMonth = `with`(YEAR, year)
+
+  def withMonth(month: Int): YearMonth = `with`(MONTH_OF_YEAR, month)
+
+  def plus(amount: Long, unit: TemporalUnit): YearMonth = unit match {
+    case MONTHS    => plusMonths(amount)
+    case YEARS     => plusYears(amount)
+    case DECADES   => plusYears(MathJDK8Bridge.multiplyExact(amount, 10))
+    case CENTURIES => plusYears(MathJDK8Bridge.multiplyExact(amount, 100))
+    case MILLENNIA => plusYears(MathJDK8Bridge.multiplyExact(amount, 1000))
+
+    case ERAS      =>
+      val era = getLong(ERA)
+      `with`(ERA, MathJDK8Bridge.addExact(era, amount))
+
+    case _: ChronoUnit =>
+      throw new UnsupportedTemporalTypeException("Unsupported unit: " + unit)
+
+    case _ => unit.addTo(this, amount)
+  }
+
+  def plusYears(years: Long): YearMonth = {
+    val newYear = YEAR.checkValidIntValue(year + years)
+    if (newYear == year) this
```

Comment:
`newYear == year` is equivalent to `years == 0` which implies that `newYear` does not need to be computed for the `if` branch, push it into the `else` branch.